app/models/scenario/user_updates.rb

```diff
@@ -93,7 +93,7 @@ def user_values_as_yaml=(values)
   def inputs
     @inputs ||=
       combined_values.map { |key, _| Input.get(key) }.
-      compact.sort_by(&:priority).reverse
+      compact.sort_by { |input| [-input.priority, input.key] }
```

Comment:
I think @ploh's aim was that two inputs with identical priorities would always execute in the same order, regardless of the order in which they are specified in `user_values`.